### DISCUSSION TITLE
Thoughts of including mason-null-ls.nvim to LazyVim

### DISCUSSION THREAD

**Discussion by florence-wolfe** [LazyVim/LazyVim#408] [Ideas]
Hey everyone,

I recently set up [mason-null-ls.nvim](https://github.com/jay-babu/mason-null-ls.nvim) in my LazyVim config and have found it to have greatly simplified both my `mason` and `null-ls` configurations.

For those that are unfamiliar with the project:
> mason-null-ls.nvim closes some gaps that exist between mason.nvim and null-ls.

At a high-level the configuration allows you to provide the `ensure_installed` table that you would in `mason`. This will install the sources in mason as you would normally expect but _also_ setup a default configuration for those same sources in `null-ls`.

`mason-null-ls` also has a `setup_handlers` method that allows you to configure and fine tune your sources if needed.

For example, I don't want the eslint formatter in my configuration but want to keep the diagnostics and code actions. My configuration would look like this.
```lua
      require("mason-null-ls").setup_handlers({
        function(source_name, methods)
          -- all sources with no handler get passed here

          -- To keep the original functionality of `automatic_setup = true`,
          -- please add the below.
          require("mason-null-ls.automatic_setup")(source_name, methods)
        end,
        eslint_d = function()
          null_ls.register(null_ls.builtins.diagnostics.eslint_d)
          null_ls.register(null_ls.builtins.code_actions.eslint_d)
        end,
      })
```

When trying to configure both null-ls and mason's sources separately I was having a lot of difficulty keeping track of the configurations. I'm of the mind that this simplifies everything quite a bit for both LazyVim and any end-user.

Would love to start a discussion and see what other people think on this

**Comment by dpetka2001** [comment]
See [this](https://github.com/LazyVim/LazyVim/discussions/390).

**Comment by djsnipa1** [comment]
> For those that are unfamiliar with the project:
> 
> > mason-null-ls.nvim closes some gaps that exist between mason.nvim and null-ls.

Thank you for posting this. I’m unfamiliar with this plugin and also have had a lot of trouble getting it all set-up. LazyVim is great and is my favorite “config framework” (for lack of a better term) but I’m having trouble getting the last 20% of my setup working correctly. Hopefully this will help.
